### PR TITLE
[DOCS] Adds elser service to the inference tutorial

### PR DESCRIPTION
**Pull request by szabosteve**
## Overview

This PR expands the Inference tutorial with the steps for using the ELSER service.

### Preview

[Inference tutorial](https://elasticsearch_bk_111406.docs-preview.app.elstc.co/guide/en/elasticsearch/reference/master/semantic-search-inference.html)

### CODE REVIEW

File: docs/reference/search/search-your-data/semantic-search-inference.asciidoc [hunk 1]
```diff
@@ -40,17 +39,15 @@ include::{es-ref-dir}/tab-widgets/inference-api/infer-api-task-widget.asciidoc[]
 ==== Create the index mapping
 
 The mapping of the destination index - the index that contains the embeddings that the model will create based on your input text - must be created.
-The destination index must have a field with the <<dense-vector, `dense_vector`>>
-field type to index the output of the used model.
+The destination index must have a field with the <<dense-vector, `dense_vector`>> field type for most models and the <<sparse-vector, `sparse_vector`>> field type for the `elser` service to index the output of the used model.
```
Comment (kderusso) [inline docs/reference/search/search-your-data/semantic-search-inference.asciidoc:42]: Nit: Sparse vector can be used with any sparse embeddings, not just ELSER.

File: docs/reference/search/search-your-data/semantic-search-inference.asciidoc [hunk 2]
```diff
@@ -103,8 +100,8 @@ POST _tasks/<task_id>/_cancel
 ==== Semantic search
 
 After the data set has been enriched with the embeddings, you can query the data using {ref}/knn-search.html#knn-semantic-search[semantic search].
-Pass a
-`query_vector_builder` to the k-nearest neighbor (kNN) vector search API, and provide the query text and the model you have used to create the embeddings.
+In case of dense vector models, pass a `query_vector_builder` to the k-nearest neighbor (kNN) vector search API, and provide the query text and the model you have used to create the embeddings.
+In case of a sparse vector model like ELSER, use a `text_expansion` query, and provide the query text with the model you have used to create the embeddings.
```
Comment (kderusso) [inline docs/reference/search/search-your-data/semantic-search-inference.asciidoc:104]: Note: `text_expansion` has been deprecated as of 8.15, and we should guide people to use `sparse_vector` queries instead.

File: docs/reference/tab-widgets/inference-api/infer-api-search.asciidoc [hunk 3]
```diff
@@ -72,6 +72,68 @@ query from the `cohere-embeddings` index sorted by their proximity to the query:
...
 // end::cohere[]
 
+// tag::elser[]
+
+[source,console]
+--------------------------------------------------
+GET elser-embeddings/_search
+{
+  "query":{
+    "text_expansion":{
```
Comment (kderusso) [inline docs/reference/tab-widgets/inference-api/infer-api-search.asciidoc:82]: Please replace this example with a `sparse_vector` query.

**Review by kderusso — CHANGES_REQUESTED** [review]
Hi, overall these docs look good but it's referring to deprecated query syntax, please make sure to update it. Thank you!

**Review by maxhniebergall — APPROVED** [review]
Looks great Istvan! This is a huge improvement, amazing!

Kathleen has some good comments which I think should be addressed.

I don't think it needs to go in this PR, but I think we should add semantic text to this guide ASAP, since it is available in serverles, and we should make a plan to replace the ingest pipeline with semantic text once it becomes available in a release.

**Comment by szabosteve** [comment]
@maxhniebergall We have a standalone [`semantic_text`](https://www.elastic.co/guide/en/elasticsearch/reference/master/semantic-search-semantic-text.html) tutorial that covers the steps of performing search with semantic_text. I connected the inference tutorial and the semantic_text tutorial with links but AFAIK we keep the inference workflow documentation at lease a little while so users can find support if they want to use the inference workflow for more complex cases.

File: docs/reference/search/search-your-data/semantic-search-inference.asciidoc (hunk 1 above)
Comment (szabosteve) [reply docs/reference/search/search-your-data/semantic-search-inference.asciidoc:42]: I made the text clearer. Our only service that uses sparse vectors currently is ELSER, and I wanted to highlight it in the tutorial so people don't need to ponder if they use the correct field type or not.

File: docs/reference/tab-widgets/inference-api/infer-api-search.asciidoc (hunk 3 above)
Comment (szabosteve) [reply docs/reference/tab-widgets/inference-api/infer-api-search.asciidoc]: Updated and tested. Changes have been pushed via [d40646](https://github.com/elastic/elasticsearch/pull/111406/commits/5d406464b6ff5924ccc506fb1118fd1f5c72b93c).

**Review by kderusso — APPROVED** [review]
Thanks for iterating, LGTM!

**Pull request merged**